api/src/importers/bcogc/base-record-utils.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+const mongoose = require('mongoose');
+const defaultLog = require('../../utils/logger')('bcogc-csv-base-record-utils');
+const RecordController = require('../../controllers/record-controller');
+const MiscConstants = require('../../utils/constants/misc');
+
+/**
+ * Ogc csv base record type handler that can be used directly, or extended if customizations are needed.
+ *
+ * @class BaseRecordUtils
+ */
+class BaseRecordUtils {
+  /**
+   * Creates an instance of BaseRecordUtils.
+   *
+   * @param {*} auth_payload user information for auditing
+   * @param {*} recordType an item from record-type-enum.js -> RECORD_TYPE
+   * @param {*} csvRow an array containing the values from a single csv row.
+   * @memberof BaseRecordUtils
+   */
+  constructor(auth_payload, recordType, csvRow) {
+    if (!recordType) {
+      throw Error('BaseRecordUtils - required recordType must be non-null.');
+    }
+
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+    this.csvRow = csvRow;
+  }
+
+  /**
+   * Transform an bcogc-csv row into a NRPTI record.
+   *
+   * Note: Only transforms common fields found in ALL supported bcogc-csv types.
+   *       To include other values, extend this class and adjust the object returned by this function as needed.
+   *
+   * @param {object} csvRow bcogc-csv row (required)
+   * @returns {object} NRPTI record.
+   * @throws {Error} if record is not provided.
+   * @memberof BaseRecordUtils
+   */
+  transformRecord(csvRow) {
+    if (!csvRow) {
+      throw Error('transformRecord - required csvRow must be non-null.');
+    }
+
+    return {
+      _schemaName: this.recordType._schemaName,
+      recordType: this.recordType.displayName,
+
+      sourceSystemRef: 'bcogc-csv'
+    };
+  }
+
+  /**
+   * Searches for an existing master record, and returns it if found.
+   *
+   * @param {*} nrptiRecord
+   * @returns {object} existing NRPTI master record, or null if none found or _sourceRefOgcInspectionId &&
+   * _sourceRefOgcDeficiencyId are null
+   * @memberof BaseRecordUtils
+   */
+  async findExistingRecord(nrptiRecord) {
+    if (!nrptiRecord._sourceRefOgcInspectionId && !nrptiRecord._sourceRefOgcDeficiencyId) {
```

Comment:
Just confirming that this is indeed an "and" rather than "or"?

Reply:
Actually, now that I think about it more, the DeficiencyId null check should be removed.

The way to uniquely identify the records is from either:
- InspectionId=`uniqueString` && DeficiencyId=`null`
- InspecitonId=`nonUniqueString` && DeficiencyId=`uniqeStringForThisInspectionId`

So we really only care that InspectionId is not null.

Reply:
Though, its an extreme edge case, as the only issue with the current setup is if InspectionId is null and DeficiencyId is not null, but that shouldn't ever be the case.  Ill update it anyways though since it is technically safer.